roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -287,280 +542,229 @@ def __init__(self, filepath, optical_channel_name=None,
+                                  description='x,y location of centroid of the roi in image_mask')
+                    ps.add_column(name='Accepted',
+                                  description='1 if ROi was accepted or 0 if rejected as a cell during segmentation operation')
+                for num, row in enumerate(roi_ids): #Expects the existing ps to be a prior nwbsegext saved nwb file with existing columns
+                    ps.add_row(id=row, image_mask=image_masks[:, :, num],
+                               RoiCentroid=roi_locations[num,:],
+                               Accepted=accepted_ids[num])
+
+                # Fluorescence Traces:
+                if 'Flourescence' not in ophys.data_interfaces:
+                    fluorescence = Fluorescence()
+                    ophys.add_data_interface(fluorescence)
+                else:
+                    fluorescence = ophys.data_interfaces['Fluorescence']
+                roi_response_dict = segext_obj.get_traces_dict()
+                roi_table_region = ps.create_roi_table_region(description=f'region for Imaging plane{plane_no_loop}',
+                                                              region=list(range(segext_obj.get_num_rois())))
+                rate = np.float('NaN') if segext_obj.get_sampling_frequency() is None else segext_obj.get_sampling_frequency()
+                for i, j in roi_response_dict.items():
+                    data = getattr(segext_obj, f'_roi_response_{i}')
+                    if data is not None:
+                        trace_name = 'RoiResponseSeries' if i == 'raw' else i.capitalize()
+                        trace_name = trace_name if plane_no_loop==0 else trace_name+f'_Plane{plane_no_loop}'
+                        input_kwargs = dict(name=trace_name, data=data.T, rois=roi_table_region, rate=rate)
+                        if trace_name not in fluorescence.roi_response_series:
+                            fluorescence.create_roi_response_series(**input_kwargs)
+
+                # create Two Photon Series:
+                if 'TwoPhotonSeries' not in nwbfile.acquisition:
+                    warn('could not find TwoPhotonSeries, using ImagingExtractor to create an nwbfile')
+
+                # adding images:
+                images_dict = segext_obj.get_images_dict()
+                images_name = 'SegmentationImages' if plane_no_loop==0 else f'SegmentationImages_Plane{plane_no_loop}'
+                if images_name not in ophys.data_interfaces:
+                    images = Images(images_name)
+                    for img_name, img_no in images_dict.items():
+                        if img_no is not None:
+                            images.add_image(GrayscaleImage(name=img_name, data=img_no))
+                    ophys.add(images)
+
+            # saving NWB file:
             io.write(nwbfile)
 
         # test read
-        with NWBHDF5IO(savepath, 'r') as io:
+        with NWBHDF5IO(save_path, 'r') as io:
```

Comment:
I think this is not necessary. We'll have specific tests in to do it.

Reply:
actually, I'd like to keep this in for now. I've had problems previously where a file is written in pynwb but cannot be read. I think we have fixed all the places where this might occur now, but I'd like to catch this as early as possible if it occurs again.